patches/server/0019-Block-server-side-chunkbans.patch

```diff
@@ -0,0 +1,165 @@
+         ConnectionProtocol connectionProtocol = channelHandlerContext.channel().attr(Connection.ATTRIBUTE_PROTOCOL).get();
+@@ -36,35 +67,68 @@ public class PacketEncoder extends MessageToByteEncoder<Packet<?>> {
+                 throw new IOException("Can't serialize unregistered packet");
+             } else {
+                 FriendlyByteBuf friendlyByteBuf = new FriendlyByteBuf(byteBuf);
+-                friendlyByteBuf.writeVarInt(integer);
+-                friendlyByteBuf.adventure$locale = channelHandlerContext.channel().attr(PaperAdventure.LOCALE_ATTRIBUTE).get(); // Paper
+-
++                // Scissors start
++                int k;
+                 try {
+-                    int i = friendlyByteBuf.writerIndex();
+-                    packet.write(friendlyByteBuf);
+-                    int j = friendlyByteBuf.writerIndex() - i;
+-                    if (j > 8388608) {
+-                        throw new IllegalArgumentException("Packet too big (is " + j + ", should be less than 8388608): " + packet);
+-                    }
++                    k = tryWrite(packet, friendlyByteBuf, channelHandlerContext, integer);
+                 } catch (Throwable var9) {
+-                    LOGGER.error("Packet encoding of packet ID {} threw (skippable? {})", integer, packet.isSkippable(), var9); // Paper - WHAT WAS IT? WHO DID THIS TO YOU? WHAT DID YOU SEE?
+-                    if (packet.isSkippable()) {
+-                        throw new SkipPacketException(var9);
+-                    } else {
+-                        throw var9;
++                    packet = capPacket(packet, integer);
++                    if (packet == null) {
++                        throw new SkipPacketException(new IllegalArgumentException("Packet too big: " + packet));
+                     }
++                    friendlyByteBuf.clear();
++                    k = tryWrite(packet, friendlyByteBuf, channelHandlerContext, integer);
+                 }
+-
+-                // Paper start
+-                int packetLength = friendlyByteBuf.readableBytes();
+-                if (packetLength > MAX_PACKET_SIZE) {
+-                    throw new PacketTooLargeException(packet, packetLength);
+-                }
+-                // Paper end
++                // Scissors end
+             }
+         }
+     }
+ 
++    // Scissors start
++    private static Packet capPacket(Packet packet, int i)
++    {
++        if (packet instanceof ClientboundBlockEntityDataPacket blockEntityDataPacket)
++        {
++            packet = new ClientboundBlockEntityDataPacket(blockEntityDataPacket.getPos(), blockEntityDataPacket.getType(), new CompoundTag());
++        }
++        else if (packet instanceof ClientboundLevelChunkPacket chunkPacket)
++        {
++            chunkPacket.clearNBT();
++        }
++        else if (packet instanceof ClientboundSetEntityDataPacket)
++        {
++            return null; // Skip
```

Comment:
Any particular reason for this being skipped?

Reply:
That's how ayunami's patch did it, it would clear the packetItems list resulting in an empty array. I could make it write each item to a buffer and check if that is too large, but I didn't do that because I thought it would be too complex and would take too many resources.

Reply:
Ah, I see.